src/middlewared/middlewared/plugins/system.py

```diff
@@ -568,11 +573,47 @@ async def info(self):
             'license': await self.middleware.run_in_thread(self._get_license),
             'boottime': datetime.fromtimestamp(psutil.boot_time()),
             'datetime': datetime.utcnow(),
-            'timezone': (await self.middleware.call('datastore.config', 'system.settings'))['stg_timezone'],
+            'birthday': birthday_date,
+            'timezone': timezone_setting,
             'system_manufacturer': manufacturer,
             'ecc_memory': ecc_memory,
         }
 
+    async def _convert_datetime(self, response, timezone_setting=timezone.utc):
+        return datetime.fromtimestamp(response.tx_time, timezone_setting)
+
+    # Sync the clock
+    @private
+    async def sync_clock(self):
+        client = ntplib.NTPClient()
+        clock = None
+        # Tries to get default ntpd server
+        try:
+            response = client.request("localhost")
+            if response.version:
+                clock = await self._convert_datetime(response)
+        except Exception:
+            # Cannot connect to NTP server
+            pass
+
+        if clock is not None:
+            return clock
+
+        # If it fails, tries the list of default ntp servers configurable
```

Comment:
We are already querying local ntp which has the ntp servers configured, I dont really see the point of doing these things below?

Reply:
Make sense.